include/drivers/st25r3911b_interrupt.h

```diff
@@ -0,0 +1,159 @@
+#define ST25R3911B_IRQ_MASK_WT              (0x080000UL)
+#define ST25R3911B_IRQ_MASK_WAM             (0x040000UL)
+#define ST25R3911B_IRQ_MASK_WPH             (0x020000UL)
+#define ST25R3911B_IRQ_MASK_WCAP            (0x010000UL)
+
+/**
+ * @}
+ */
+
+/** @brief Initialize NFC Reader interrupts.
+ *
+ *  @details The NFC Reader informs about interruptions by
+ *           a rising edge on the pin. This function initializes
+ *           the GPIO interface to handle this.
+ *
+ *  @param[in] irq_sem Semaphore needed to synchronize
+ *                     processing interrupts from other context.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_irq_init(struct k_sem *irq_sem);
+
+/** @brief Modify NFC Reader interrupts.
+ *
+ *  @details This function modifies the currently enabled interrupts.
+ *           The set mask covers the clear mask, which means that first,
+ *           interrupts from the clear mask are disabled, and then interrupts
+ *           from the set mask are enabled. The mask can be composed of several
+ *           interrupts. For example: <tt>mask = ST25R3911B_IRQ_MASK_PAR |
+ *           ST25R3911B_IRQ_MASK_CRC</tt>
+ *
+ *           See @ref st25r3911b_irq_def for available interrupts.
+ *
+ *  @param[in] clr_mask Interrupts clear mask.
+ *  @param[in] set_mask Interrupts set mask.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_irq_modify(u32_t clr_mask, u32_t set_mask);
+
+/** @brief Enable NFC Reader interrupts.
+ *
+ *  @details The interrupts mask can be composed of several interrupts
+ *           (see @ref st25r3911b_irq_def).
+ *
+ *  @param[in] mask Interrupts to enable mask.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_irq_enable(u32_t mask);
+
+/** @brief Disable NFC Reader interrupts.
+ *
+ *  @details The interrupts mask can be composed of several interrupts
+ *           (see @ref st25r3911b_irq_def).
+ *
+ *  @param[in] mask Interrupts to disable mask.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_irq_disable(u32_t mask);
+
+/** @brief Clear NFC Reader interrupts status.
+ *
+ *  @details Interrupts are cleared by reading NFC Reader
+ *           interrupts registers.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r39_irq_clear(void);
+
+/** @brief Wait for NFC Reader interrupts.
+ *
+ *  @details This function can wait for one interrupt or
+ *           for several (defined by a mask).
+ *
+ *  @param[in] mask Interrupts mask (see @ref st25r3911b_irq_def).
+ *  @param[in] timeout Wait time in milliseconds.
+ *
+ *  @return Interrupts status, reading after time-out or when
```

Comment:
I don't understand the return value.

Reply:
This mean tha you get a value, which tells you which interrupts are triggered.

Reply:
OK. Moved the clarification about when the function returns to the description.